keras_cv/layers/preprocessing/random_resized_crop.py

```diff
@@ -0,0 +1,244 @@
+    random height and width, and then resize them. To do this, we first sample a
+    random value for area using `crop_area_factor` and a value for aspect ratio using
+    `aspect_ratio_factor`. Further we get the new height and width by
+    dividing and multiplying the old height and width by the random area
+    respectively. We then sample offsets for height and width and clip them such
+    that the cropped area does not exceed image boundaries. Finally we do the
+    actual cropping operation and resize the image to `target_size`.
+
+    Args:
+        target_size: A tuple of two integers used as the target size to ultimately crop
+            images to.
+        crop_area_factor: A tuple of two floats, ConstantFactorSampler or
+            UniformFactorSampler. The ratio of area of the cropped part to
+            that of original image is sampled using this factor. Represents the
+            lower and upper bounds for the area relative to the original image
+            of the cropped image before resizing it to `target_size`.  For
+            self-supervised pretraining a common value for this parameter is
+            `(0.08, 1.0)`.  For fine tuning and classification a common value for this
+            is `0.8, 1.0`.
+        aspect_ratio_factor: A tuple of two floats, ConstantFactorSampler or
+            UniformFactorSampler. Aspect ratio means the ratio of width to
+            height of the cropped image. In the context of this layer, the aspect ratio
+            sampled represents a value to distort the aspect ratio by.
+            Represents the lower and upper bound for the aspect ratio of the
+            cropped image before resizing it to `target_size`.  For most tasks, this
+            should be `(3/4, 4/3)`.  To perform a no-op provide the value `(1.0, 1.0)`.
+        interpolation: (Optional) A string specifying the sampling method for
+            resizing. Defaults to "bilinear".
+        seed: (Optional) Used to create a random seed. Defaults to None.
+    """
+
+    def __init__(
+        self,
+        target_size,
+        crop_area_factor,
+        aspect_ratio_factor,
+        interpolation="bilinear",
+        seed=None,
+        **kwargs,
+    ):
+        super().__init__(seed=seed, **kwargs)
+
+        self._check_class_arguments(target_size, crop_area_factor, aspect_ratio_factor)
+
+        if isinstance(aspect_ratio_factor, tuple):
+            max_aspect_ratio = max(aspect_ratio_factor)
+        elif isinstance(aspect_ratio_factor, core.FactorSampler):
+            pass
+        else:
+            raise ValueError(
+                "Expected `aspect_ratio` to be tuple or FactorSampler. Received "
+                f"aspect_ratio_factor={aspect_ratio_factor}."
+            )
+
+        self.target_size = target_size
+        self.aspect_ratio_factor = preprocessing.parse_factor(
+            aspect_ratio_factor,
+            max_value=max_aspect_ratio,
```

Comment:
max_value should be a constant that is used for validation.  Check out how it is used inside of parse_factor.

The goal of max_value is to prevent users from passing values outside of the value range.

Reply:
In this case, please use min_value=0, max_value=None.

Reference: https://github.com/keras-team/keras-cv/blob/master/keras_cv/utils/preprocessing.py#L109